spring-integration-mongodb/src/main/java/org/springframework/integration/mongodb/store/MessageDocument.java

```diff
@@ -151,6 +154,15 @@ public Object getGroupId() {
 		return this.groupId;
 	}
 
+	@Nullable
+	public String getCondition() {
+		return this.condition;
+	}
+
+	public void setCondition(String condition) {
+		this.condition = condition;
+	}
+
```

Comment:
Are there any backwards compatibility issues (existing documents)?

Reply:
Existing documents comes without a `condition` field: nothing to read - nothing to map.
It is really a trick of all of these NO-SQL DBs: no schema - can change data structure any time!